query-connector/flyway/sql/V02_01__add_fhir_servers.sql

```diff
@@ -0,0 +1,11 @@
+CREATE TABLE IF NOT EXISTS fhir_servers (
+    id UUID DEFAULT uuid_generate_v4 (),
+    name TEXT,
+    hostname TEXT,
+    headers JSON DEFAULT NULL
```

Comment:
Will we need to add columns for authentication creds for non-public servers, or will we just store a token in the headers column?

Reply:
I believe we'll just store them in the headers column unless you think there's a good reason to keep it separate. We have some OAUTHSCOPES in the headers for the CernerHelios connection but we haven't had any username, etc. authentication yet.

```
function configureEHX(xdestination: string): DevFhirServerConfig {
  const headers = {
    Accept: "application/json, application/*+json, */*",
    "Accept-Encoding": "gzip, deflate, br",
    "Content-Type": "application/fhir+json; charset=UTF-8",
    "X-DESTINATION": xdestination,
    "X-POU": "PUBHLTH",
    "X-Request-Id": uuidv4(),
    prefer: "return=representation",
    "Cache-Control": "no-cache",
  };
  if (xdestination === "CernerHelios") {
    (headers as Record<string, string>)["OAUTHSCOPES"] =
      "system/Condition.read system/Encounter.read system/Immunization.read system/MedicationRequest.read system/Observation.read system/Patient.read system/Procedure.read system/MedicationAdministration.read system/DiagnosticReport.read system/RelatedPerson.read";
  }
  return {
    id: xdestination,
    name: xdestination,
    hostname: "https://concept01.ehealthexchange.org:52780/fhirproxy/r4/",
    headers,
    trustSelfSigned: true,
  };
}
```